src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.cs

```diff
@@ -54,24 +54,58 @@ public async Task<ImmutableArray<CodeAction>> AddConstructorParametersFromMember
             }
         }
 
+        /// <summary>
+        /// Try to find a constructor in <paramref name="containingType"/> whose parameters is the subset of <paramref name="parameters"/> by comparing name.
+        /// If multiple constructors meet the condition, the one with more parameters will be returned.
+        /// It will not consider those constructors as potential candidates
+        /// 1. Constructor with empty parameter list.
+        /// 2. Constructor's parameter list contains 'ref' or 'params'
+        /// </summary>
+        protected override IMethodSymbol GetDelegatedConstructor(
+            INamedTypeSymbol containingType,
+            ImmutableArray<IParameterSymbol> parameters)
+        {
+            return containingType.InstanceConstructors
+                .WhereAsArray(constructor => IsParamtersContainedInConstructor(constructor, parameters.SelectAsArray(p => p.Name)))
```

Comment:
1. Paramters
2. Don't use .WhereAsArray if you're not planning on using the returned array.  `.Where` is sufficient and desirable here.

Reply:
.SelectAsArray is realizing hte names *every* time you check a constructor.  pull that out above into a local and use hte local.